engine/pkg/client/internal/endpoint/endpoint.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 PingCAP, Inc.
+// Copyright 2021 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
```

Comment:
Can add more unit tests to this package later.

Reply:
This file is copied from Etcd's repo. I will look into this later.